CI: smoke test failure in docker compose up

<!-- You have to provide the following information whenever possible. -->

**Describe what's wrong**
<!-- A clear and concise description of what works not as it is supposed to. -->
https://github.com/timeplus-io/proton/actions/runs/7147196986/job/19467073281#step:7:394
https://github.com/timeplus-io/proton/actions/runs/7147196986/job/19467577389#step:7:396
https://github.com/timeplus-io/proton/actions/runs/7146988181/job/19466812153#step:7:402
https://github.com/timeplus-io/proton/actions/runs/7146988163/job/19466818559#step:7:404
**Error message and/or stacktrace**
<!-- If applicable, add screenshots to help explain your problem. -->
```
Building wheels for collected packages: dockerpty, docopt
  Building wheel for dockerpty (pyproject.toml): started
  Building wheel for dockerpty (pyproject.toml): finished with status 'done'
  Created wheel for dockerpty: filename=dockerpty-0.4.1-py3-none-any.whl size=16597 sha256=5c8cdb6b0ca75c778d1d0f77d324f8d653db5b451ffd54af52e800d9daf40d94
  Stored in directory: /root/.cache/pip/wheels/18/00/32/f75cd03098074f988a01c59a2e3a55ae9c0773eb66acb4cb5e
  Building wheel for docopt (pyproject.toml): started
  Building wheel for docopt (pyproject.toml): finished with status 'done'
  Created wheel for docopt: filename=docopt-0.6.2-py2.py3-none-any.whl size=13706 sha256=76d1ed94a03bf61ee77974d9bb54d700417830b130a9a7f1c7ecca9ab384da3e
  Stored in directory: /root/.cache/pip/wheels/fc/ab/d4/5da2067ac95b36618c629a5f93f809425700506f72c9732fac
Successfully built dockerpty docopt
Installing collected packages: unidiff, texttable, sseclient-py, pytz, makefun, kafka-python, docopt, urllib3, tzlocal, toml, six, python-dotenv, pyrsistent, pycparser, py, psycopg2-binary, pluggy, packaging, loguru, jmespath, iniconfig, idna, distro, decopatch, charset-normalizer, certifi, bcrypt, attrs, websocket-client, requests, python-dateutil, pytest-cases, pytest, proton-driver, jsonschema, dockerpty, cffi, timeplus, pytest-metadata, pynacl, docker, cryptography, botocore, s3transfer, pytest-html, pyOpenSSL, paramiko, boto3, docker-compose
Successfully installed attrs-23.1.0 bcrypt-4.1.1 boto3-1.20.4 botocore-1.23.54 certifi-2023.11.17 cffi-1.16.0 charset-normalizer-3.3.2 cryptography-39.0.2 decopatch-1.4.10 distro-1.8.0 docker-7.0.0 docker-compose-1.29.2 dockerpty-0.4.1 docopt-0.6.2 idna-3.6 iniconfig-2.0.0 jmespath-0.10.0 jsonschema-3.2.0 kafka-python-2.0.2 loguru-0.7.2 makefun-1.15.2 packaging-23.2 paramiko-3.3.1 pluggy-1.3.0 proton-driver-0.2.10 psycopg2-binary-2.9.3 py-1.11.0 pyOpenSSL-23.0.0 pycparser-2.21 pynacl-1.5.0 pyrsistent-0.20.0 pytest-6.2.5 pytest-cases-3.6.5 pytest-html-3.1.1 pytest-metadata-2.0.4 python-dateutil-2.8.2 python-dotenv-0.21.1 pytz-2023.3.post1 requests-2.31.0 s3transfer-0.5.2 six-1.16.0 sseclient-py-1.8.0 texttable-1.7.0 timeplus-1.1.2 toml-0.10.2 tzlocal-5.2 unidiff-0.7.0 urllib3-1.26.7 websocket-client-0.59.0
2023-12-08 23:54:46,903 [    INFO] [MainProcess] [ci_runner] [<module>] docker compose up... (ci_runner.py:488)
2023-12-08 23:54:46,903 [   DEBUG] [MainProcess] [ci_runner] [<module>] env_docker_compose_res: False (ci_runner.py:489)
Traceback (most recent call last):
  File "/actions-runner/_work/proton/proton/tests/stream/ci_runner.py", line 491, in <module>
ci_runner starting: run_mode = github, loop = 1, logging_level=DEBUG, test_retry = True, test_suite_timeout = 900, test_folders = test_stream_smoke starts
args_dict = {'run_mode': 'github', 'debug': True, 'test_folders': 'test_stream_smoke', 'test_suites': 'None', 'id': 'None', 'loop': 1, 'settings': 'redp,nativelog', 'cluster_query_route_mode': 'None', 'cluster_query_node': 'None', 'create_stream_shards': 'None', 'create_stream_replicas': 'None', 'no_retry': False, 'test_suite_timeout': 900, 'tmp': 'None'}
    raise Exception("Env docker compose up failure.")
Exception: Env docker compose up failure.
self._event.value = {'event_type': 'test_event', 'payload': {'detailed_type': 'status', 'details': 'start'}}
self._tag_value = {'version': '0.1', 'repo_name': 'proton', 'test_info': {'test_id': '2a6c0e00-9625-11ee-b524-9113bba36471', 'test_name': 'ci_runner', 'test_type': 'ci_smoke'}, 'build_info': {'build_type': 'release_build', 'pr_number': 'develop', 'commit_sha': '2bd0eddb3b8735d018bbf73c21c27b8718902e12'}, 'run_time_info': {'os': 'Linux', 'platform': 'X64'}}
test_event_start sent
Error: Process completed with exit code 1.
```
**Additional context**
<!--Add any other context about the problem here. -->

--- a/.github/workflows/manual_trigger_test.yml
+++ b/.github/workflows/manual_trigger_test.yml
@@ -69,6 +69,10 @@ jobs:
         # FIXME: remove this line after pyyaml community fixed install bug
         pip install pyyaml==5.3.1
 
+        # FIXME(yokofly): docker 7.0.0 introduce a breaking change
+        # https://github.com/docker/docker-py/issues/3194
+        pip install docker==6.1.3
+
         pip install -r helpers/requirements.txt
 
         python ci_runner.py --settings=redp,nativelog
@@ -158,6 +162,10 @@ jobs:
         
         # FIXME: remove this line after pyyaml community fixed install bug
         pip install pyyaml==5.3.1
+
+        # FIXME(yokofly): docker 7.0.0 introduce a breaking change
+        # https://github.com/docker/docker-py/issues/3194
+        pip install docker==6.1.3
 
         pip install -r helpers/requirements.txt
 

--- a/.github/workflows/nightly_test.yml
+++ b/.github/workflows/nightly_test.yml
@@ -102,6 +102,10 @@ jobs:
             # FIXME: remove this line after pyyaml community fixed install bug
             pip install pyyaml==5.3.1
 
+            # FIXME(yokofly): docker 7.0.0 introduce a breaking change
+            # https://github.com/docker/docker-py/issues/3194
+            pip install docker==6.1.3
+
             pip install -r helpers/requirements.txt
             python ci_runner.py --debug --settings=redp,nativelog
 
@@ -211,6 +215,10 @@ jobs:
 
             # FIXME: remove this line after pyyaml community fixed install bug
             pip install pyyaml==5.3.1
+
+            # FIXME(yokofly): docker 7.0.0 introduce a breaking change
+            # https://github.com/docker/docker-py/issues/3194
+            pip install docker==6.1.3
 
             pip install -r helpers/requirements.txt
             python ci_runner.py --debug --settings=redp,nativelog

--- a/.github/workflows/proton_ci.yml
+++ b/.github/workflows/proton_ci.yml
@@ -7,7 +7,6 @@ on: # yamllint disable-line rule:truthy
       - reopened
       - opened
       - ready_for_review
-      - labeled
     branches:
       - develop
     paths-ignore:
@@ -21,8 +20,8 @@ jobs:
   verify:
     name: Labels check
     runs-on: ubuntu-latest
-    # Either the PR is not a draft OR it has the label 'pr-test'
-    if: github.event.pull_request.draft == false || contains(github.event.pull_request.labels.*.name, 'pr-test')
+    # CI is triggered for untrusted contributors only when the `pr-test` label is present and a new commit is pushed.
+    if: github.event.pull_request.draft == false
     env:
       GH_PERSONAL_ACCESS_TOKEN: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
     steps:

--- a/.github/workflows/proton_ci.yml
+++ b/.github/workflows/proton_ci.yml
@@ -164,6 +164,10 @@ jobs:
         # FIXME: remove this line after pyyaml community fixed install bug
         pip install pyyaml==5.3.1
 
+        # FIXME(yokofly): docker 7.0.0 introduce a breaking change
+        # https://github.com/docker/docker-py/issues/3194
+        pip install docker==6.1.3
+
         pip install -r helpers/requirements.txt
         df -h
         python ci_runner.py --settings=redp,nativelog

--- a/tests/proton_ci/proton_helper.py
+++ b/tests/proton_ci/proton_helper.py
@@ -65,7 +65,7 @@ def prepare_event(test_results, event_type, **optional_event_msg):
             if "OK" != test_status:
                 test_result_flag = test_result_flag * 0
             detailed_summary.append(current_row)
-        event['test_result'] = "seccess" if test_result_flag else "failed"
+        event['test_result'] = "success" if test_result_flag else "failed"
         if optional_event_msg is not None:
             event.update(optional_event_msg)
         event['payload'] = detailed_summary

--- a/tests/proton_ci/run_check.py
+++ b/tests/proton_ci/run_check.py
@@ -34,6 +34,7 @@ TRUSTED_CONTRIBUTORS = {
         "qijun-niu-timeplus",
         "lizhou1111",
         "leo-cai-timeplus",
+        "ilselintp",
     ]
 }
 

